BusinessLayer/JsonMessageBuilder/JsonMessageBuilder.cpp

```diff
@@ -9,11 +9,47 @@
 #include "DataLayer/BatteryData/I_BatteryData.h"
 #include "DataLayer/MpptData/I_MpptData.h"
 #include "DataLayer/LightsData/I_LightsData.h"
+#include "DataLayer/AuxBmsData/I_AuxBmsData.h"
 
 JsonMessageBuilder::JsonMessageBuilder()
 {
 }
 
+QJsonObject JsonMessageBuilder::buildAuxBmsMessage(const I_AuxBmsData& data)
+{
+    QJsonObject auxBmsJson = QJsonObject();
+
+    switch (data.getPrechargeState())
+    {
+        case I_AuxBmsData::PrechargeState::IDLE:
+            auxBmsJson[JsonFormat::PRECHARGE_STATE] = "IDLE";
```

Comment:
These are still using the old values of precharge state. We are expecting values of OFF, COMMON_ENGAGED, CHARGE_ENGAGED, ALL_ENGAGED as according to the protocol. Sorry for not catching this sooner.